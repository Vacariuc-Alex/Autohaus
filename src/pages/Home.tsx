import {useEffect, useState} from "react";
import CompanyFilter from "../components/CompanyFilter";
import InfoCard from "../components/InfoCard";
import ContentCanvas from "../utils/styledComponents/ContentCanvas";
import Flex from "../utils/styledComponents/Flex";
import useAxios from "../utils/hooks/useAxios";
import PageSelector from "../components/PageSelector";
import BasicPagination from "../components/BasicPagination";

const Home = () => {

    type Product = {
        id: string,
        company: string,
        model: string,
        year: number,
        vin: string,
        color: string,
        price: number
    };

    const [products, setProducts] = useState<Product[]>([]);
    const [companies, setCompanies] = useState<string[]>([]);
    const [elementsPerPage, setElementsPerPage] = useState<number>(10);
    const [numberOfPages, setNumberOfPages] = useState<number>(0);
    const [currentPage, setCurrentPage] = useState<number>(1);

    const {response, error, loading} = useAxios({
        method: 'get',
        url: '/products'
    });

    const filterProductsByCompanies = (responseData: Product[], companies: string[]) => {
        return companies.length !== 0
            ? responseData.filter((e: Product) => companies.includes(e.company))
            : responseData;
    }

    const calculateNumberOfPages = (sortedProducts: Product[], elementsPerPage: number) => {
        return Math.ceil(sortedProducts.length / elementsPerPage)
    };

    const limitSortedProducts  = (i: number, j: number, sortedProducts: Product[], elementsPerPage: number) => {
        return j < elementsPerPage && j < sortedProducts.length - (i * elementsPerPage)
    };
    
    const displayElements = () => {
        if (response !== null) {
            const responseData = response.data;
            let sortedProducts: Product[] = filterProductsByCompanies(responseData, companies);
            const calculatedPages = calculateNumberOfPages(sortedProducts, elementsPerPage);
            let pagedProducts: Product[][] = [];

            setNumberOfPages(calculatedPages);
            for (let i = 0; i < calculatedPages; i++) {
                pagedProducts[i] = [];
                for (let j = 0; limitSortedProducts(i, j, sortedProducts, elementsPerPage); j++) {
                    pagedProducts[i][j] = sortedProducts[i * elementsPerPage + j];
                }
            }
            setProducts(pagedProducts[currentPage - 1]);
        }
    }

    useEffect(() => {
        displayElements();
    }, [companies, currentPage, elementsPerPage, response]);

    const handleElementsPerPageChange = (e: number) => {
        setCurrentPage(1);
        setElementsPerPage(e);
    };

    const handleCurrentPage = (e: number) => {
        setCurrentPage(e);
    };

    const handleCompanies = (e: string[]) => {
        setCurrentPage(1);
        setCompanies(e);
    }

    if (error) {
        return (
            <h1>Error: {error}</h1>
        );
    } else if (loading) {
        return (
            <h1>Loading...</h1>
        );
    }

    return (
        <>
            <Flex>
                <ContentCanvas>
                    {
                        products.map((e, i) => (
                            <InfoCard elementProperties={e} key={i}/>
                        ))
                    }
                </ContentCanvas>
                <CompanyFilter companiesProp={handleCompanies}/>
            </Flex>
            <PageSelector onElementsPerPageChangeProp={handleElementsPerPageChange}/>
            <BasicPagination numberOfPagesProp={numberOfPages} currentPageProp={handleCurrentPage}/>
        </>
    );
}

export default Home;
