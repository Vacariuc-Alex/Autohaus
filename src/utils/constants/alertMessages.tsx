export const PRODUCT_ADDED_SUCCESSFULLY = "The product has been successfully added!";
export const FEATURE_BLOCKED =
    "Oops! It looks like this feature is currently unavailable.\n" +
    "We encountered an unexpected error on our server.\n" +
    "Please try refreshing the page or reaching out to our support team for assistance.";
export const FAILED_REQUEST =
    "We're sorry, but we couldn't fulfill your request due to server issues.\n" +
    "The request failed with the following reason:";
export const PRODUCT_UPDATED_SUCCESSFULLY = "The product has been successfully modified!";
export const PRODUCT_DELETED_SUCCESSFULLY = "The product has been successfully removed!";
export const UNKNOWN_REQUEST_TYPE =
    "We're sorry, but we couldn't fulfill your request due to an unexpected error";
