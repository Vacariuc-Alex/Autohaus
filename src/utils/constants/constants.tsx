export type Product = {
    id: string;
    company: string;
    model: string;
    year: number;
    vin: string;
    color: string;
    price: number;
    owner: User;
    images?: string[];
    isFavourite?: boolean;
};

export type UserRole = "ADMIN" | "USER";

export type User = {
    id: string,
    username: string,
    email: string,
    password: string,
    role: UserRole,
    avatar?: string
};

export type UserLogin = {
    username: string;
    password: string;
}

export const userAvatarOptions = {
    wishlist: "Wishlist",
    login: "Login",
    logout: "Logout"
}

export const EMPTY_STRING = "";
export const DEFAULT_ROLE = "USER";
export const EXPIRATION_PREFIX = "__expiration__";
export const AUTHENTICATED_USER = "authenticatedUser";
export const ZERO = "0";
export const ONE = "1";
export const DEFAULT_EXPIRATION_PERIOD = 24;
