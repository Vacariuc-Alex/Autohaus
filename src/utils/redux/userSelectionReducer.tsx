import {createSlice, PayloadAction} from "@reduxjs/toolkit";

type DataState = {
    selectedCompanies: string[];
    currentPage: number;
    numberOfPages: number;
    elementsPerPage: number;
}

const initialState: DataState = {
    selectedCompanies: [],
    currentPage: 1,
    numberOfPages: 1,
    elementsPerPage: 10
}

export const userSelectionSlice = createSlice({
    name: 'userSelectionSlice',
    initialState,
    reducers: {
        // selectedCompanies
        addNewCompany: (state: DataState, action: PayloadAction<string>) => {
            const selectedCompanies = state.selectedCompanies;
            if (!selectedCompanies.includes(action.payload)) {
                state.selectedCompanies = [...selectedCompanies, action.payload]
            }
        },
        removeExistingCompany: (state: DataState, action: PayloadAction<string>) => {
            const index = state.selectedCompanies.indexOf(action.payload);
            if (index !== -1) {
                state.selectedCompanies.splice(index, 1);
            }
        },

        // currentPage
        setCurrentPage: (state: DataState, action: PayloadAction<number>) => {
            state.currentPage = action.payload;
        },
        resetCurrentPage: (state: DataState) => {
            state.currentPage = 1;
        },

        // numberOfPages
        setNumberOfPages: (state: DataState, action: PayloadAction<number>) => {
            state.numberOfPages = action.payload;
        },
        resetNumberOfPages: (state: DataState) => {
            state.numberOfPages = 1;
        },

        // elementsPerPage
        setElementsPerPage: (state: DataState, action: PayloadAction<number>) => {
            state.elementsPerPage = action.payload;
        }
    }
});

export const {
    addNewCompany,
    removeExistingCompany,
    setCurrentPage,
    resetCurrentPage,
    setNumberOfPages,
    resetNumberOfPages,
    setElementsPerPage
} = userSelectionSlice.actions;
export default userSelectionSlice.reducer;
