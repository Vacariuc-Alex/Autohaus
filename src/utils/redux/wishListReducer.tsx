import {createSlice, PayloadAction} from '@reduxjs/toolkit'

type DataState = {
    ids: number[];
}

const initialState: DataState = {
    ids: [],
}

export const wishListSlice = createSlice({
    name: 'wishListSlice',
    initialState,
    reducers: {
        addItem: (state: DataState, action: PayloadAction<number>) => {
            if (!state.ids.includes(action.payload)) {
                state.ids.push(action.payload);
            }
        },
        removeItem: (state: DataState, action: PayloadAction<number>) => {
            const index = state.ids.findIndex((e) => e === action.payload);
            if (index !== -1) {
                state.ids.splice(index, 1);
            }
        }
    }
});

export const {
    addItem,
    removeItem
} = wishListSlice.actions;
export default wishListSlice.reducer;
