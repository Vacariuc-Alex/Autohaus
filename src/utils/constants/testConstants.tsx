import {Product} from "./constants";

//Mock store
export type InitialState = {
    wishListStore: {
        ids: number[]
    },
    productsStore: {
        responseData: Product[];
        loading: boolean;
        error: string | null;
        isRequestExecuted: boolean;
    }
};

export const initialState: InitialState = {
    wishListStore: {
        ids: []
    },
    productsStore: {
        responseData: [
            {
                id: 1,
                company: "WAZ",
                model: "2106",
                year: 1986,
                vin: "JTDZN3EU4E3035524",
                color: "Red",
                price: 250,
                isFavourite: false
            },
            {
                id: 2,
                company: "UAZ",
                model: "452",
                year: 1965,
                vin: "WAURGAFD1DN635608",
                color: "Green",
                price: 100,
                isFavourite: false
            },
        ],
        loading: false,
        error: "",
        isRequestExecuted: true
    }
};
