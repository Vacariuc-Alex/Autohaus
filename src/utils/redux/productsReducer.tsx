import {createSlice, PayloadAction} from "@reduxjs/toolkit";
import {Product} from "src/utils/constants/constants";

type DataState = {
    responseData: Product[];
    loading: boolean;
    error: string | null;
    isRequestExecuted: boolean;
}

const initialState: DataState = {
    responseData: [],
    loading: false,
    error: null,
    isRequestExecuted: false
};

const productsSlice = createSlice({
    name: "productsSlice",
    initialState,
    reducers: {
        fetchDataRequest(state: DataState) {
            state.loading = true;
            state.error = null;
        },
        fetchDataSuccess(state: DataState, action: PayloadAction<Product[]>) {
            state.responseData = action.payload;
            state.isRequestExecuted = true;
            state.loading = false;
        },
        fetchDataError(state: DataState, action: PayloadAction<string>) {
            state.error = action.payload;
            state.loading = false;
        }
    }
});

export const {
    fetchDataRequest,
    fetchDataSuccess,
    fetchDataError
} = productsSlice.actions;
export default productsSlice.reducer;
