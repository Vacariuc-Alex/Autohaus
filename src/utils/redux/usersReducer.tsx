import {createSlice, PayloadAction} from "@reduxjs/toolkit";
import {AUTHENTICATED_USER, DEFAULT_EXPIRATION_PERIOD, User, UserLogin} from "src/utils/constants/constants";
import {
    deleteItemFromLocalStorage,
    setItemInLocalStorageWithExpiration
} from "src/utils/helpers/sessionStorageWithExpiration";
import {WRONG_CREDENTIALS} from "src/utils/constants/errorMessages";
import {encryptText} from "src/utils/helpers/encryption";

// Initial data
type DataState = {
    responseData: User[];
    loading: boolean;
    error: string | null;
    isRequestExecuted: boolean;
}

const initialState: DataState = {
    responseData: [],
    loading: false,
    error: null,
    isRequestExecuted: false,
};

// Slice
const usersSlice = createSlice({
    name: "usersSlice",
    initialState,
    reducers: {
        fetchUserDataRequest(state: DataState) {
            state.loading = true;
            state.error = null;
        },
        fetchUserDataSuccess(state: DataState, action: PayloadAction<User[]>) {
            state.responseData = action.payload;
            state.isRequestExecuted = true;
            state.loading = false;
        },
        fetchUserDataError(state: DataState, action: PayloadAction<string>) {
            state.error = action.payload;
            state.loading = false;
        },
        authenticateUser(state: DataState, action: PayloadAction<UserLogin>) {
            const {username, password} = action.payload;
            const user = state.responseData
                .find((e) => e.username === username && e.password === encryptText(password));

            if(user) {
                setItemInLocalStorageWithExpiration(AUTHENTICATED_USER, user, DEFAULT_EXPIRATION_PERIOD);
            } else {
                throw new Error(WRONG_CREDENTIALS);
            }
        },
        logoutUser() {
            deleteItemFromLocalStorage(AUTHENTICATED_USER);
        }
    }
});

export const {
    fetchUserDataRequest,
    fetchUserDataSuccess,
    fetchUserDataError,
    authenticateUser,
    logoutUser,
} = usersSlice.actions;
export default usersSlice.reducer;
