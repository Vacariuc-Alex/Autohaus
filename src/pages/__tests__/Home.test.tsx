import {cleanup, render, screen, within} from "@testing-library/react";
import Home from "src/pages/Home";
import React from "react";
import {InitialState, initialState} from "src/utils/constants/testConstants";
import productsReducer from "src/utils/redux/productsReducer";
import wishListReducer from "src/utils/redux/wishListReducer";
import {configureStore} from "@reduxjs/toolkit";
import {Provider} from "react-redux";
import {deepCopy} from "src/utils/helpers/deepCopy";
import {
    APP_BAR,
    BOX,
    BOX_COMPONENT,
    CARD,
    CONTENT_CANVAS, ERROR,
    FLEX,
    PAGINATION_STACK,
    RIGHT_PANEL
} from "src/utils/constants/dataTestIds";
import userSelectionReducer, {addNewCompany, removeExistingCompany} from "src/utils/redux/userSelectionReducer";
import usersReducer from "src/utils/redux/usersReducer";

//Mock store
const mockStore = configureStore({
    reducer: {
        wishListStore: wishListReducer,
        usersStore: usersReducer,
        productsStore: productsReducer,
        userSelectionStore: userSelectionReducer
    },
    preloadedState: initialState,
});

//Mock useNavigate
const mockedNavigate = jest.fn();
jest.mock("react-router-dom", () => ({
    ...jest.requireActual("react-router-dom"),
    useNavigate: () => mockedNavigate
}));

describe("Home component", () => {
    afterEach(() => {
        cleanup();
    });

    describe("Should render Home component", () => {
        test("When no actions are performed", () => {
            render(
                <Provider store={mockStore}>
                    <Home/>
                </Provider>
            );
            const navbar = screen.getByTestId(APP_BAR);
            const flex = screen.getByTestId(FLEX);
            const contentCanvas = within(flex).getByTestId(CONTENT_CANVAS);
            const cards = within(contentCanvas).getAllByTestId(CARD);
            const rightPanel = within(flex).getByTestId(RIGHT_PANEL);
            const boxComponent = screen.getByTestId(BOX_COMPONENT);
            const paginationStack = screen.getByTestId(PAGINATION_STACK);

            expect(navbar).toBeInTheDocument();
            expect(flex).toBeInTheDocument();
            expect(contentCanvas).toBeInTheDocument();
            expect(cards.length).toBe(2);
            expect(rightPanel).toBeInTheDocument();
            expect(boxComponent).toBeInTheDocument();
            expect(paginationStack).toBeInTheDocument();
        });

        test("When a company is selected", () => {
            mockStore.dispatch(addNewCompany("UAZ"))
            render(
                <Provider store={mockStore}>
                    <Home/>
                </Provider>
            );
            const navbar = screen.getByTestId(APP_BAR);
            const flex = screen.getByTestId(FLEX);
            const contentCanvas = within(flex).getByTestId(CONTENT_CANVAS);
            const cards = within(contentCanvas).getAllByTestId(CARD);
            const rightPanel = within(flex).getByTestId(RIGHT_PANEL);
            const boxComponent = screen.getByTestId(BOX_COMPONENT);
            const paginationStack = screen.getByTestId(PAGINATION_STACK);

            expect(navbar).toBeInTheDocument();
            expect(flex).toBeInTheDocument();
            expect(contentCanvas).toBeInTheDocument();
            expect(cards.length).toBe(1);
            expect(rightPanel).toBeInTheDocument();
            expect(boxComponent).toBeInTheDocument();
            expect(paginationStack).toBeInTheDocument();
        });
        mockStore.dispatch(removeExistingCompany("UAZ"))
    });

    describe("Verify request execution with loading and error states", () => {
        test("Should display Loading component", () => {
            const copyInitialState: InitialState = deepCopy(initialState);
            copyInitialState.productsStore.isRequestExecuted = false;
            copyInitialState.productsStore.loading = true;
            const copyMockStore = configureStore({
                reducer: {
                    wishListStore: wishListReducer,
                    usersStore: usersReducer,
                    productsStore: productsReducer,
                    userSelectionStore: userSelectionReducer
                },
                preloadedState: copyInitialState
            });

            render(
                <Provider store={copyMockStore}>
                    <Home/>
                </Provider>
            );
            const loading = screen.getByTestId(BOX);

            expect(loading).toBeInTheDocument();
        });

        test("Should display error message when usersStore error", () => {
            const copyInitialState: InitialState = deepCopy(initialState);
            copyInitialState.usersStore.error = "Mock error";
            const copyMockStore = configureStore({
                reducer: {
                    wishListStore: wishListReducer,
                    usersStore: usersReducer,
                    productsStore: productsReducer,
                    userSelectionStore: userSelectionReducer
                },
                preloadedState: copyInitialState
            });

            render(
                <Provider store={copyMockStore}>
                    <Home/>
                </Provider>
            );
            const error = screen.getByTestId(ERROR);

            expect(error).toBeInTheDocument();
            expect(error).toHaveTextContent(`Error: ${copyInitialState.usersStore.error}`);
        });

        test("Should display error message when productsStore error", () => {
            const copyInitialState: InitialState = deepCopy(initialState);
            copyInitialState.productsStore.error = "Mock error";
            const copyMockStore = configureStore({
                reducer: {
                    wishListStore: wishListReducer,
                    usersStore: usersReducer,
                    productsStore: productsReducer,
                    userSelectionStore: userSelectionReducer
                },
                preloadedState: copyInitialState
            });

            render(
                <Provider store={copyMockStore}>
                    <Home/>
                </Provider>
            );
            const error = screen.getByTestId(ERROR);

            expect(error).toBeInTheDocument();
            expect(error).toHaveTextContent(`Error: ${copyInitialState.productsStore.error}`);
        });

        test("Should display error message when productsStore and usersStore error", () => {
            const copyInitialState: InitialState = deepCopy(initialState);
            copyInitialState.productsStore.error = "Mock error";
            copyInitialState.usersStore.error = "Mock error";
            const copyMockStore = configureStore({
                reducer: {
                    wishListStore: wishListReducer,
                    usersStore: usersReducer,
                    productsStore: productsReducer,
                    userSelectionStore: userSelectionReducer
                },
                preloadedState: copyInitialState
            });

            render(
                <Provider store={copyMockStore}>
                    <Home/>
                </Provider>
            );
            const error = screen.getByTestId(ERROR);

            expect(error).toBeInTheDocument();
            expect(error).toHaveTextContent(
                `Error: ${copyInitialState.productsStore.error}, ${copyInitialState.usersStore.error}`);
        });
    });
});
