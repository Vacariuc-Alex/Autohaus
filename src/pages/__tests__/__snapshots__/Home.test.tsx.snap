// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component Should match snapshot 1`] = `
Array [
  <header
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-zphf2y-MuiPaper-root-MuiAppBar-root"
    data-testid="app-bar"
  >
    <div
      className="MuiContainer-root MuiContainer-maxWidthXl css-km5s8k-MuiContainer-root"
      data-testid="container"
    >
      <div
        className="MuiToolbar-root MuiToolbar-regular css-r6ewbb-MuiToolbar-root"
        data-testid="toolbar"
      >
        <svg
          data-testid="fa-car-icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
              "display": "flex",
              "marginRight": 30,
              "transform": "scale(1.5)",
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M499.99 176h-59.87l-16.64-41.6C406.38 91.63 365.57 64 319.5 64h-127c-46.06 0-86.88 27.63-103.99 70.4L71.87 176H12.01C4.2 176-1.53 183.34.37 190.91l6 24C7.7 220.25 12.5 224 18.01 224h20.07C24.65 235.73 16 252.78 16 272v48c0 16.12 6.16 30.67 16 41.93V416c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32v-32h256v32c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32v-54.07c9.84-11.25 16-25.8 16-41.93v-48c0-19.22-8.65-36.27-22.07-48H494c5.51 0 10.31-3.75 11.64-9.09l6-24c1.89-7.57-3.84-14.91-11.65-14.91zm-352.06-17.83c7.29-18.22 24.94-30.17 44.57-30.17h127c19.63 0 37.28 11.95 44.57 30.17L384 208H128l19.93-49.83zM96 319.8c-19.2 0-32-12.76-32-31.9S76.8 256 96 256s48 28.71 48 47.85-28.8 15.95-48 15.95zm320 0c-19.2 0-48 3.19-48-15.95S396.8 256 416 256s32 12.76 32 31.9-12.8 31.9-32 31.9z"
          />
        </svg>
        <a
          className="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-1kkti6u-MuiTypography-root"
          data-testid="typography"
          href="#"
        >
          Autohaus
        </a>
        <div
          className="MuiBox-root css-1l0r181"
          data-testid="box"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-zmvk00-MuiButtonBase-root-MuiButton-root"
            data-testid="button"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            Home
          </button>
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-zmvk00-MuiButtonBase-root-MuiButton-root"
            data-testid="button"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            News
          </button>
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-zmvk00-MuiButtonBase-root-MuiButton-root"
            data-testid="button"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            Wishlist
          </button>
        </div>
      </div>
    </div>
  </header>,
  <div
    className="sc-dmyCSP goQLtE"
    data-testid="content-canvas"
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <div
      className="sc-eDLKkx hexZKI"
      data-testid="no-item-container"
      style={
        Object {
          "opacity": 0,
        }
      }
    >
      <img
        alt="Sad Heart"
        data-testid="img-container"
        src="SadHeart.png"
        style={
          Object {
            "marginLeft": "calc((100% - 300px) / 2)",
            "width": 300,
          }
        }
      />
      <h3
        className="MuiTypography-root MuiTypography-h3 MuiTypography-gutterBottom css-1mktp8m-MuiTypography-root"
        data-testid="typography-heading-container"
      >
        You don't have any favourite items!
      </h3>
      <p
        className="MuiTypography-root MuiTypography-body1 css-ix1f03-MuiTypography-root"
        data-testid="typography-paragraph-container"
      >
        You can add an item to favourite by clicking on the heart that shows up when you hover over the picture!
      </p>
    </div>
  </div>,
]
`;
